cilium/cmd/policy_wait.go

```diff
@@ -48,30 +48,44 @@ var policyWaitCmd = &cobra.Command{
 
 			needed := len(eps)
 			ready := 0
+			notReady := 0
 
 			for _, ep := range eps {
-				if ep.Policy != nil && ep.PolicyRevision >= reqRevision &&
-					ep.State == models.EndpointStateReady {
-					ready++
+				if ep.Policy != nil {
+					if ep.PolicyRevision >= reqRevision &&
+						ep.State == models.EndpointStateReady {
+						ready++
+					} else if ep.State == models.EndpointStateNotReady {
+						notReady++
+					}
 				}
 			}
 
 			if ready == needed {
-				if haveWaited {
+				if timeWaited > 0 {
 					fmt.Printf("\n")
 				}
 				return
+			} else if timeWaited > failWaitTime && notReady > 0 {
+				fmt.Printf("\n")
+				// Fail earlier if any endpoints have a failed state
+				Fatalf("%d endpoints have failed regeneration after %d seconds\n", notReady, timeWaited)
+			} else if timeWaited > maxWaitTime {
+				fmt.Printf("\n")
+				// Fail after timeout
+				Fatalf("%d endpoints still not ready after %d seconds (%d failed)\n", needed-ready, timeWaited, notReady)
 			}
-
 			fmt.Printf("\rWaiting for endpoints to run policy revision %d: %d/%d              ",
 				reqRevision, ready, needed)
 			time.Sleep(time.Duration(waitTime) * time.Second)
-			haveWaited = true
+			timeWaited += waitTime
 		}
 	},
 }
 
 func init() {
 	policyCmd.AddCommand(policyWaitCmd)
 	policyWaitCmd.Flags().IntVar(&waitTime, "sleep-time", 1, "Sleep interval between checks (seconds)")
+	policyWaitCmd.Flags().IntVar(&failWaitTime, "fail-wait-time", 60, "Wait time after which command fails if endpoint regeration fails (seconds)")
+	policyWaitCmd.Flags().IntVar(&maxWaitTime, "wait-time", 360, "Wait time after which command fails (seconds)")
```

Comment:
Perhaps this should be `max-wait-time` to be more clear about its interaction with `fail-wait-time`

Reply:
OK.